Lib/test/test_tempfile.py

```diff
@@ -84,8 +88,13 @@ def nameCheck(self, name, dir, pre, suf):
         nsuf  = nbase[len(nbase)-len(suf):]
 
         if dir is not None:
-            self.assertIs(type(name), str if type(dir) is str else bytes,
-                          "unexpected return type")
+            self.assertIs(
+                type(name),
+                str
+                if type(dir) is str or isinstance(dir, os.PathLike) else
```

Comment:
Maybe break this up to it's own line since its getting a bid crowded as a ternary